### PR TITLE
fix: Fixed middleware stack

### PR DESCRIPTION
**Pull request by aemr3**
Middlewares are not working with the latest updates. I added a test to demonstrate the problem. 

Fixed it by re-passing user_middleware attribute to cloned_app. Btw, if you add middlewares to versioned app it was working. But if you add middlewares before versioning the app it wasn't working. 

I suspect there maybe other issues with dependency_overrides or exception_handlers. Not sure yet.

cc: @alexschimpf 
(Sorry for the commit messages not following your linting standards)

### CODE REVIEW

File: fastapi_versionizer/versionizer.py [hunk 1]
```diff
@@ -327,6 +327,7 @@ def _clone_app(app: FastAPI) -> FastAPI:
             try:
                 cloned_app = app.__class__(**kwargs)
                 cloned_app.router.lifespan_context = app.router.lifespan_context
+                cloned_app.user_middleware = app.user_middleware
```
Comment (alexschimpf) [inline fastapi_versionizer/versionizer.py:330]: Yeah, I was afraid of this kind of thing. Most of the field names match the constructor's kwarg names. The assumption does not hold for this case (field name is `user_middleware` but kwarg name is `middleware`). 

I might end up refactoring all this cloning a little bit... possibly going back to modifying the FastAPI object in place.

**Review by alexschimpf — APPROVED** [review]

**Comment by alexschimpf** [comment]
@aemr3 I made a PR that goes back to modifying the FastAPI object in place, instead of trying to hackily clone it. This should ensure that all properties of your FastAPI are properly preserved.

https://github.com/alexschimpf/fastapi-versionizer/pull/43/files

**Comment by aemr3** [comment]
> @aemr3 I made a PR that goes back to modifying the FastAPI object in place, instead of trying to hackily clone it. This should ensure that all properties of your FastAPI are properly preserved.
> 
> https://github.com/alexschimpf/fastapi-versionizer/pull/43/files

Cool, thanks. I will test it soon. It looks better this way.

**Pull request merged**